### DISCUSSION TITLE
How to download large file with progress?

### DISCUSSION THREAD

**Discussion by nebulabox** [awslabs/aws-sdk-swift#679] [Q&A]
It is impossible to download large files directly into memory. A progress callback is required at the same time. 

I may have missed some details, how should I use this library to download large files with progress update?

**Comment by ganeshnj** [comment]
That's a fair use case, we have [plan](https://github.com/awslabs/aws-sdk-swift/issues/685) to add progress tracking to download (GetObject) and upload (PutObject) operations.

Apart from progress tracking, is there anything that you are looking and is not supported by SDK?

I would suggest checking [GetObject](https://github.com/awslabs/aws-sdk-swift/blob/0.4.0/release/AWSS3/S3ClientProtocol.swift#L562) operation which seems like your use case.

**Comment by MainasuK** [comment]
I hope I can do something like this to make nest progress tracking easy.

```swift
myProgress.addChild(theGetObjectInput.progress, withPendingUnitCount: myProgress.totalUnitCount)
```
